test/integration/helpers/project.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+var h = require('../helpers.js');
+
+// TODO: factor this out into a proper page object
+exports.visitCreatePage = function() {
+  h.goToPage('create-project');
+};
+
+exports.projectDetails = function() {
+  var timestamp = (new Date()).getTime();
+  var project = {
+    name:        'console-test-project-' + timestamp,
+    displayName: 'Console integration test Project ' + timestamp,
+    description: 'Created by integration tests'
+  };
+  return project;
+};
+
+exports.createProject = function(project, uri) {
+  for (var key in project) {
+    h.setInputValue(key, project[key]);
+  }
+  h.clickAndGo('Create', uri);
+};
+
+exports.deleteProject = function(project) {
+  h.goToPage('/');
+  var projectTile = element(by.cssContainingText(".project-info", project['name']));
+  projectTile.element(by.css('.fa-trash-o')).click();
+  h.setInputValue('confirmName', project.name);
+  var deleteButton = element(by.cssContainingText(".modal-dialog .btn", "Delete"));
+  browser.wait(protractor.ExpectedConditions.elementToBeClickable(deleteButton), 2000);
+  deleteButton.click();
+  h.waitForPresence(".alert-success", "marked for deletion");
+};
+
+exports.deleteAllProjects = function() {
```

Comment:
We ran in to issues getting deleteAllProjects() to run at teardown, and ended up using deleteAllProjects() in an afterEach() inside each suite.  We literally spent hours trying to figure out how to make this all work before arriving at this as a compromise.  If you know of a better way, by all means...